libibcm/cm.h

```diff
@@ -325,8 +325,8 @@ struct ib_cm_attr_param {
 int ib_cm_attr_id(struct ib_cm_id *cm_id,
 		  struct ib_cm_attr_param *param);
 
-#define IB_CM_ASSIGN_SERVICE_ID_MASK htobe64(0xFF00000000000000ULL)
-#define IB_CM_ASSIGN_SERVICE_ID      htobe64(0x0200000000000000ULL)
+#define IB_CM_ASSIGN_SERVICE_ID_MASK cpu_to_be64(0xFF00000000000000ULL)
+#define IB_CM_ASSIGN_SERVICE_ID      cpu_to_be64(0x0200000000000000ULL)
 
```

Comment:
I have reworked <infiniband/endian.h> such that it redefines the glibc macros instead of introducing new conversion functions. Additionally, all #include <infiniband/endian.h> directives have been changed into #include <endian.h>.